app/src/main/java/com/ncnf/database/DatabaseService.java

```diff
@@ -1,125 +1,245 @@
+            if(task.isSuccessful()){
+                List<T> result = new ArrayList<>();
+                for(DocumentSnapshot document :  task.getResult().getDocuments()){
+                    result.add((T) registry.get(collectionType).toObject(document.getId(), document.getData()));
+                }
+                futureResponse.complete(result);
+            } else {
+                futureResponse.completeExceptionally(task.getException());
+            }
+        });
 
         return futureResponse;
     }
 
-    public CompletableFuture<DatabaseResponse> updateArrayField(String path, String arrayField, Object value){
-        return this.updateField(path, arrayField, FieldValue.arrayUnion(value));
-    }
+    @Override
+    public <T> CompletableFuture<List<T>> withFieldLike(String collectionPath, String field, String value, Class<T> collectionType) {
+        CompletableFuture<List<T>> futureResponse = new CompletableFuture<>();
+
+        this.db.collection(collectionPath).orderBy(field).startAt(value).endAt(value + "\uf8ff").get().addOnCompleteListener(task -> {
+            if(task.isSuccessful()){
+                List<T> result = new ArrayList<>();
+                for(DocumentSnapshot document : task.getResult().getDocuments()){
+                    result.add((T) registry.get(collectionType).toObject(document.getId(), document.getData()));
+                }
+
+                futureResponse.complete(result);
+            } else {
+                futureResponse.completeExceptionally(task.getException());
+            }
+        });
 
-    public CompletableFuture<DatabaseResponse> removeArrayField(String path, String arrayField, Object value){
-        return this.updateField(path, arrayField, FieldValue.arrayRemove(value));
+        return futureResponse;
     }
 
-    public <U> CompletableFuture<DatabaseResponse> whereEqualTo(String path, String field, List<U> values) {
-        if(values.size() == 0) throw new IllegalArgumentException("You must pass at least one value.");
+    @Override
+    public <T, R> CompletableFuture<List<R>> whereArrayContains(String collectionPath, String field, T value, Class<R> type) {
+        CompletableFuture<List<R>> futureResponse = new CompletableFuture<>();
 
-        CompletableFuture<DatabaseResponse> futureResponse = new CompletableFuture<>();
-        CollectionReference ref = this.db.collection(path);
-        Query query = ref.whereEqualTo(field, values.get(0));
-        for (int i = 1; i < values.size(); ++i) {
-            query = query.whereEqualTo(field, values.get(i));
-        }
-        query.get().addOnCompleteListener(task -> onTaskComplete(task, futureResponse));
+        this.db.collection(collectionPath).whereArrayContains(field, value).get().addOnCompleteListener(task -> {
+            if(task.isSuccessful()){
+                List<R> result = new ArrayList<>();
+                for(DocumentSnapshot document : task.getResult().getDocuments()){
```

Comment:
Minor detail, when you do this, unless the compiler has optimizations to avoid this, the .getResult().getDocuments() is performed at each iteration of the loop